lib/index.js

```diff
@@ -1,32 +1,33 @@
-var canvas = document.getElementById('game');
+var canvas = document.getElementById('puts-puts-golf');
 var context = canvas.getContext('2d');
 var Hole = require('./hole')
 var Ball = require("./ball");
 var Bumper = require("./bumper");
+var Game = require("./game");
 
-var golfBall = new Ball(100, 400, 6, context);
-var obstacleOne = new Bumper(50, 50, 250, 60, context);
-var obstacleTwo = new Bumper(50, 450, 250, 460, context);
-var obstacleThree = new Bumper(50, 50, 60, 450, context);
-var obstacleFour = new Bumper(250, 50, 260, 460, context);
-var bumperOne = new Bumper(150, 170, 155, 220, context, "blue");
-var obstacles = [obstacleOne, obstacleTwo, obstacleThree, obstacleFour, bumperOne]
-var puttHole = new Hole(150, 100, context)
+var game = new Game(context);
 
+requestAnimationFrame(startLoop);
 
-requestAnimationFrame(function gameLoop() {
+function startLoop() {
+  var newGame = new Game(context);
+  requestAnimationFrame(gameLoop)
+};
+
+function gameLoop() {
   context.beginPath();
-  // context.clearRect(0, 0, canvas.width, canvas.height);
+  context.clearRect(0, 0, canvas.width, canvas.height);
   context.closePath();
-  puttHole.draw();
-  golfBall.draw();
-  golfBall.holeCheck(puttHole)
-  if(golfBall.moving) {
-      golfBall.move()
+  game.hole.draw();
+  game.ball.draw();
+  game.ball.holeCheck(game.hole)
+  if(game.ball.moving) {
+      game.ball.move()
   }
- requestAnimationFrame(gameLoop);
-  obstacles.forEach(function(obstacle) { obstacle.draw();})
-});
+
+  game.bumpers.forEach(function(bumper) { bumper.draw();})
+  requestAnimationFrame(gameLoop);
+};
 
 canvas.addEventListener('mousedown', function(event) {
```

Comment:
@hhoopes this event listener is not storing mouse coordinates and triggering the move ball function. Can you explain the changes?